assets/js/googlesitekit/modules/datastore/sharing-settings.js

```diff
@@ -0,0 +1,247 @@
+		return API.set(
+			'core',
+			'modules',
+			'sharing-settings',
+			savedSharingSettings,
+			{
+				useCache: false,
+			}
+		);
+	},
+	reducerCallback: ( state, { settings } ) => {
+		return {
+			...state,
+			savedSharingSettings: settings,
+			sharingSettings: settings,
+		};
+	},
+	argsToParams: ( savedSharingSettings ) => ( { savedSharingSettings } ),
+	validateParams: ( { savedSharingSettings } = {} ) => {
+		invariant( savedSharingSettings, 'savedSharingSettings is required.' );
+	},
+} );
+
+const baseActions = {
+	/**
+	 * Sets the sharing settings management role of given module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} moduleSlug Module slug.
+	 * @param {string} management New management role for module, one of: all_admins, owner.
+	 * @return {Object} Action for SET_SHARING_MANAGEMENT.
+	 */
+	setSharingManagement( moduleSlug, management ) {
+		invariant( moduleSlug, 'moduleSlug is required.' );
+		const managementRoles = [ 'all_admins', 'owner' ];
+		invariant(
+			managementRoles.includes( management ),
+			`management must be one of: ${ managementRoles.join( ', ' ) }.`
+		);
+		return {
+			payload: {
+				moduleSlug,
+				management,
+			},
+			type: SET_SHARING_MANAGEMENT,
+		};
+	},
+
+	/**
+	 * Sets the sharing settings management role of given module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string}   moduleSlug Module slug.
+	 * @param {string[]} roles      List of roles the module is shared with.
+	 * @return {Object} Action for SET_SHARED_ROLES.
+	 */
+	setSharedRoles( moduleSlug, roles ) {
+		invariant( moduleSlug, 'moduleSlug is required.' );
+		invariant(
+			Array.isArray( roles ) &&
+				roles.every( ( role ) => typeof role === 'string' ),
+			'roles must be an array of strings.'
+		);
+		return {
+			payload: {
+				moduleSlug,
+				roles,
+			},
+			type: SET_SHARED_ROLES,
+		};
+	},
+
+	/**
+	 * Saves sharingSettings for dashboard sharing.
+	 *
+	 * Save sharingSettings for dashboard sharing.
+	 * Update ownerID from the response for the modules in the sharingSettings state.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {Object} Object with `{response, error}`.
+	 */
+	*saveSharingSettings() {
+		const registry = yield Data.commonActions.getRegistry();
+		// TODO: Refactor to use `getSharingSettings` selector
+		// to obtain the `sharingSettings` from the state in 4795.
+		const { sharingSettings } = registry.stores[
+			CORE_MODULES
+		].store.getState();
+
+		const {
+			response,
+			error,
+		} = yield fetchSaveSharingSettingsStore.actions.fetchSaveSharingSettings(
+			sharingSettings
+		);
+
+		// Update module owner IDs in the sharing settings modules.
+		if ( Object.keys( response?.newOwnerIDs ).length ) {
+			for ( const [ slug, ownerID ] of Object.entries(
+				response.newOwnerIDs
+			) ) {
+				const storeName = registry
+					.select( CORE_MODULES )
+					.getModuleStoreName( slug );
+
+				yield registry.dispatch( storeName ).setOwnerID( ownerID );
```

Comment:
Looks like we have a few other instances of this, but only raw actions can be yielded from a generator. All actions returned from `dispatch` are bound action creators that return a promise so this will still invoke the action, but it won't be "awaited" like a normal yielded action would be.

This should be safe to dispatch without awaiting it as it should run synchronously but something to keep in mind.

```suggestion
				registry.dispatch( storeName ).setOwnerID( ownerID );
```